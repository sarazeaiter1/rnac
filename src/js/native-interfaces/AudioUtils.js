import { Platform } from 'react-native';

import AudioRecorderManager from './AudioRecorderManager';

let AudioUtils;
const {
  MainBundlePath,
  NSCachesDirectoryPath,
  NSDocumentDirectoryPath,
  NSLibraryDirectoryPath,
  CachesDirectoryPath,
  DocumentDirectoryPath,
  LibraryDirectoryPath,
  PicturesDirectoryPath,
  MusicDirectoryPath,
  DownloadsDirectoryPath
} = AudioRecorderManager;
export const getAudioUtils = platform => {
  return platform === 'android'
    ? {
        MainBundlePath: MainBundlePath,
        CachesDirectoryPath: CachesDirectoryPath,
        DocumentDirectoryPath: DocumentDirectoryPath,
        LibraryDirectoryPath: LibraryDirectoryPath,
        PicturesDirectoryPath: PicturesDirectoryPath,
        MusicDirectoryPath: MusicDirectoryPath,
        DownloadsDirectoryPath: DownloadsDirectoryPath
      }
    : {
        MainBundlePath: MainBundlePath,
        CachesDirectoryPath: NSCachesDirectoryPath,
        DocumentDirectoryPath: NSDocumentDirectoryPath,
        LibraryDirectoryPath: NSLibraryDirectoryPath
      };
};
if (Platform.OS === 'ios') {
  AudioUtils = {
    MainBundlePath: MainBundlePath,
    CachesDirectoryPath: NSCachesDirectoryPath,
    DocumentDirectoryPath: NSDocumentDirectoryPath,
    LibraryDirectoryPath: NSLibraryDirectoryPath
  };
} else if (Platform.OS === 'android') {
  AudioUtils = {
    MainBundlePath: MainBundlePath,
    CachesDirectoryPath: CachesDirectoryPath,
    DocumentDirectoryPath: DocumentDirectoryPath,
    LibraryDirectoryPath: LibraryDirectoryPath,
    PicturesDirectoryPath: PicturesDirectoryPath,
    MusicDirectoryPath: MusicDirectoryPath,
    DownloadsDirectoryPath: DownloadsDirectoryPath
  };
}

export default AudioUtils;
