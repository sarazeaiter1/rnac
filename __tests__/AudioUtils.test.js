// import React from 'react';
// import { Platform } from 'react-native';
// import renderer from 'react-test-renderer';
// import { shallow } from 'enzyme';
import AudioUtils, {
  getAudioUtils
} from '../src/js/native-interfaces/AudioUtils';
describe('AudioUtils', () => {
  test('exports', () => {
    expect(AudioUtils).toMatchSnapshot();
  });
  test('exports getAudioUtils', () => {
    expect(getAudioUtils).toMatchSnapshot();
  });
  test('exports getAudioUtils callable', () => {
    expect(getAudioUtils()).toMatchSnapshot();
  });
  test('exports getAudioUtils callable android ', () => {
    expect(getAudioUtils('android')).toMatchSnapshot();
  });
});
