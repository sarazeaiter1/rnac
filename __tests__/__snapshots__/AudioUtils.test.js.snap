// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AudioUtils exports 1`] = `
Object {
  "CachesDirectoryPath": undefined,
  "DocumentDirectoryPath": undefined,
  "LibraryDirectoryPath": undefined,
  "MainBundlePath": undefined,
}
`;

exports[`AudioUtils exports getAudioUtils 1`] = `[Function]`;

exports[`AudioUtils exports getAudioUtils callable 1`] = `
Object {
  "CachesDirectoryPath": undefined,
  "DocumentDirectoryPath": undefined,
  "LibraryDirectoryPath": undefined,
  "MainBundlePath": undefined,
}
`;

exports[`AudioUtils exports getAudioUtils callable android  1`] = `
Object {
  "CachesDirectoryPath": undefined,
  "DocumentDirectoryPath": undefined,
  "DownloadsDirectoryPath": undefined,
  "LibraryDirectoryPath": undefined,
  "MainBundlePath": undefined,
  "MusicDirectoryPath": undefined,
  "PicturesDirectoryPath": undefined,
}
`;
